new_session_test.go

```diff
@@ -222,3 +235,85 @@ func (s *testSessionSuite) TestAutocommit(c *C) {
 	tk.MustExec("set autocommit='On'")
 	c.Assert(int(tk.Se.Status()&mysql.ServerStatusAutocommit), Greater, 0)
 }
+
+func (s *testSessionSuite) TestSchemaCheckerSQL(c *C) {
+	store, err := tikv.NewMockTikvStore()
+	c.Assert(err, IsNil)
+	tidb.SetStatsLease(0)
+	lease := 20 * time.Millisecond
+	tidb.SetSchemaLease(lease)
+	dom, err := tidb.BootstrapSession(store)
+	c.Assert(err, IsNil)
+	defer dom.Close()
+	defer store.Close()
+
+	tk := testkit.NewTestKit(c, store)
+	tk.MustExec(`use test`)
+	tk1 := testkit.NewTestKit(c, store)
+	tk1.MustExec("use test")
+
+	// create table
+	tk.MustExec(`create table t (id int, c int);`)
+	tk.MustExec(`create table t1 (id int, c int);`)
+	// insert data
+	tk.MustExec(`insert into t values(1, 1);`)
+
+	// The schema version is out of date in the first transaction, but the SQL can be retried.
+	tk.MustExec(`begin;`)
+	tk1.MustExec(`alter table t add index idx(c);`)
+	time.Sleep(lease + 2*time.Millisecond)
+	tk.MustExec(`insert into t values(2, 2);`)
```

Comment:
the old test is t1. It's better to use t, which can prove we have failed but retried without raising any error.